src/expression/definitions/within.ts

```diff
@@ -295,12 +295,24 @@ class Within implements Expression {
         if (isValue(args[1])) {
             const geojson = (args[1] as any);
             if (geojson.type === 'FeatureCollection') {
+                const polygonsCoords = [];
                 for (let i = 0; i < geojson.features.length; ++i) {
-                    const type = geojson.features[i].geometry.type;
-                    if (type === 'Polygon' || type === 'MultiPolygon') {
-                        return new Within(geojson, geojson.features[i].geometry);
+                    const {type, coordinates} = geojson.features[i].geometry;
+                    if (type === 'Polygon') {
+                        polygonsCoords.push(coordinates);
                     }
+                    if (type === 'MultiPolygon') {
+                        polygonsCoords.push(...coordinates);
```

Comment:
Seems like this adds an array of ploygons, I would expect to see `flat` somewhere around here...

Reply:
No, here polygons are pushed one by one. So it's already flat